Dark mode 🕶️ support.

biocom opened the pull request:
This PR adds Dark mode support (addressing #622 )

Todo:
- Test to make sure all colors are correct consistent. On some occasions there might be a black text where it should be a light text. (e.g. settings panel).
- The dark mode background animations will be fine-tuned in a future PR.
<img width="1494" alt="Screen Shot 2021-05-14 at 12 28 22" src="https://user-images.githubusercontent.com/31534717/118258206-e2a9fe00-b4af-11eb-84ed-5ada420314d4.png">

<img width="320" alt="Screen Shot 2021-05-14 at 12 28 37" src="https://user-images.githubusercontent.com/31534717/118258241-eb9acf80-b4af-11eb-9ef0-f99ca7ed9b68.png"><img width="320" alt="Screen Shot 2021-05-14 at 12 30 24" src="https://user-images.githubusercontent.com/31534717/118258462-2b61b700-b4b0-11eb-8435-e4b8bc283f83.png">



Expert mode enabled (Switch arrows makes place for the '+ Add Send' button:
<img width="320" alt="Screen Shot 2021-05-14 at 12 33 43" src="https://user-images.githubusercontent.com/31534717/118258882-b2169400-b4b0-11eb-857a-51581a131f57.png"><img width="320" alt="Screen Shot 2021-05-14 at 12 33 25" src="https://user-images.githubusercontent.com/31534717/118258891-b3e05780-b4b0-11eb-8f2d-b204d28cba0f.png">

anxolin reviewed (commented):
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FVp3ftHKvKpASA%2Fgiphy.gif&token=cow" data-gitmeme-token="cow"><img src="https://media1.giphy.com/media/Vp3ftHKvKpASA/giphy.gif" title="Created by gitme.me with /cow"/></a>

Awesome! I like the color selection. One early feedback, is that it feels the contrast is not so good for the title and the main text. I would whiten a bit more the text (or darken the main color, not sure, or both a little bit)

Most of the black texts are not visible now, but I'm sure this is cause this is a draft.

Lastly, the laser effect is great, but it was happening to fast imo, so it gets too much attention. I would make it more subtle


Amazing progress, people will love it.

biocom commented on src/components/AddressInputPanel/index.tsx:
```diff
@@ -98,7 +101,11 @@ export default function AddressInputPanel({
             <RowBetween>
-              <TYPE.black color={theme.text2} fontWeight={500} fontSize={14}>
+              <TYPE.black
+                // color={theme.text2}
```
Disabling the color prop, so we can control it exclusively with CSS.

W3stside replied:
mmm i know its just one change but this may cause headaches in the future as it's not clear this was modded. I would rework this in pure CSS maybe with specificity? `&&&&&&` or sth idk if that'll work

biocom commented on src/components/WalletModal/Option.tsx:
```diff
@@ -62,7 +62,7 @@ const CircleWrapper = styled.div`
 const HeaderText = styled.div`
   ${({ theme }) => theme.flexRowNoWrap};
-  color: ${props => (props.color === 'blue' ? ({ theme }) => theme.primary1 : ({ theme }) => theme.text1)};
+  /* color: ${props => (props.color === 'blue' ? ({ theme }) => theme.primary1 : ({ theme }) => theme.text1)}; */
```
Disabling the color definition as it inherits the right light/dark mode for now.

W3stside replied:
these should be modded as well

biocom commented on src/components/swap/TradePrice.tsx:
```diff
@@ -26,7 +26,7 @@ export default function TradePrice({ price, showInverted, setShowInverted }: Tra
       fontWeight={500}
       fontSize={14}
-      color={theme.text2}
+      // color={theme.text2}
```
Disabling the color prop, so we can control it exclusively with CSS.

W3stside replied:
also this needs to be modded..

biocom replied:
Reverted changes on this file.

biocom commented on src/custom/components/AccountDetails/index.tsx:
```diff
@@ -118,7 +129,7 @@ export default function AccountDetails({
 
   return (
-    <>
+    <Wrapper>
```
Adding a parent wrapper to create a parent Styled component, to then target specific child component's CSS.

biocom commented on src/custom/components/CurrencyInputPanel/CurrencyInputPanelMod.tsx:
```diff
@@ -168,13 +173,17 @@ export default function CurrencyInputPanel({
                 <TYPE.body
                   onClick={onMax}
-                  color={theme.text2}
+                  // color={theme.text2}
```
Disabling the color prop, so we can control it exclusively with CSS.

biocom replied:
Reverted to original mod.

biocom commented on src/custom/components/CurrencyInputPanel/CurrencyInputPanelMod.tsx:
```diff
@@ -168,13 +173,17 @@ export default function CurrencyInputPanel({
             <RowBetween>
-              <TYPE.body color={theme.text2} fontWeight={500} fontSize={14}>
+              <TYPE.body
+                // color={theme.text2}
```
Disabling the color prop, so we can control it exclusively with CSS.

biocom commented on src/custom/components/Header/index.tsx:
```diff
@@ -127,6 +142,9 @@ export default function Header() {
         </HeaderElement>
         <HeaderElementWrap>
+          <StyledMenuButton onClick={() => toggleDarkMode()}>
+            {darkMode ? <Moon size={20} /> : <Sun size={20} />}
```
Adding back the dark mode toggle.

biocom commented on src/custom/components/Settings/SettingsMod.tsx:
```diff
@@ -147,18 +147,18 @@ export default function SettingsTab({ className, SettingsButton }: SettingsTabPr
             <RowBetween style={{ padding: '0 2rem' }}>
               <div />
-              <Text fontWeight={500} fontSize={20}>
+              <Text fontWeight={500} fontSize={20} color={theme.text2}>
```
Some complexity around targeting the <Text> component specifically (to override/set the `color` CSS property). Just added the color prop here in the Mod file. Let me know if we need to change.

W3stside replied:
its fine

biocom commented on src/custom/components/Web3Status/Web3StatusMod.tsx:
```diff
@@ -123,11 +123,11 @@ export function newTransactionsFirst(a: TransactionDetails, b: TransactionDetail
-    🧦
-  </span>
-)
+// const SOCK = (
```
Disabling the SOCK stuff (commented out). Not in use.

W3stside replied:
i know its tempting to just knock it out here but you shouldnt do this in this PR as it has no relevance to it. It could be hard to identify exactly where this happened in the future should sth happen. For now leave it but best to create a sep PR or issue for this as it's neither related or pressing/breaking anything if it stays

biocom replied:
Reverted.

biocom commented on src/custom/components/swap/SwapModalFooter/SwapModalFooterMod.tsx:
```diff
@@ -56,7 +56,11 @@ export default function SwapModalFooter({
+          <Text
+            fontWeight={400}
+            fontSize={14}
+            // color={theme.text2}
```
Disabling the color prop

biocom replied:
Reverted.

MareenG commented:
In the About section the middle picture doesn't adjust to dark mode as the other two.
<img width="647" alt="Bildschirmfoto 2021-05-14 um 12 42 45" src="https://user-images.githubusercontent.com/34510341/118259903-01a98f80-b4b2-11eb-8f06-281bb48641e5.png">

biocom commented on src/custom/components/swap/SwapModalFooter/SwapModalFooterMod.tsx:
```diff
@@ -80,7 +84,11 @@ export default function SwapModalFooter({
+            <TYPE.black
+              fontSize={14}
+              fontWeight={400}
+              // color={theme.text2}
```
Disabling the color prop

biocom commented on src/custom/components/swap/SwapModalFooter/SwapModalFooterMod.tsx:
```diff
@@ -118,7 +126,11 @@ export default function SwapModalFooter({
+            <TYPE.black
+              fontSize={14}
+              fontWeight={400}
+              // color={theme.text2}
```
Disabling the color prop

biocom commented on src/custom/components/swap/SwapModalHeader/SwapModalHeaderMod.tsx:
```diff
@@ -61,7 +61,11 @@ export default function SwapModalHeader({
-        <ArrowDown size="16" color={theme.text2} style={{ marginLeft: '4px', minWidth: '16px' }} />
+        <ArrowDown
+          size="16"
+          // color={theme.text2}
```
Disabling the color prop

biocom commented on src/custom/components/swap/TradeSummary.tsx:
```diff
@@ -48,7 +48,7 @@ export default function TradeSummary({ trade, allowedSlippage }: { trade: TradeW
         <RowBetween>
           <RowFixed>
-            <TYPE.black fontSize={14} fontWeight={400} color={theme.text2}>
+            <TYPE.black fontSize={14} fontWeight={400} color={theme.text1}>
```
Complexity around targeting the <TYPE> by simple styled component overrides. Opting for a simple change here. If we need to do it otherwise, let me know.

biocom commented on src/custom/components/swap/TradeSummary.tsx:
```diff
@@ -77,7 +77,7 @@ export default function TradeSummary({ trade, allowedSlippage }: { trade: TradeW
         <RowBetween>
           <RowFixed>
-            <TYPE.black fontSize={14} fontWeight={400} color={theme.text2}>
+            <TYPE.black fontSize={14} fontWeight={400} color={theme.text1}>
```
Complexity around targeting the <TYPE> by simple styled component overrides. Opting for a simple change here. If we need to do it otherwise, let me know.

biocom commented on src/custom/pages/Swap/SwapMod.tsx:
```diff
@@ -365,7 +365,7 @@ export default function Swap({
       <AppBody className={className}>
         <SwapHeader />
-        <Wrapper id="swap-page">
+        <Wrapper id="swap-page" className={isExpertMode ? 'expertMode' : ''}>
```
Classname used to apply conditional styling. Alternative is passing it as a prop to the styled component. This seemed the easiest/least complex.

biocom commented on src/custom/pages/Swap/SwapMod.tsx:
```diff
@@ -471,7 +471,11 @@ export default function Swap({
+                      <Text
+                        fontWeight={500}
+                        fontSize={14}
+                        // color={theme.text2}
```
Disabling the color prop.

biocom commented on src/custom/pages/Swap/SwapMod.tsx:
```diff
@@ -483,10 +487,20 @@ export default function Swap({
+                      <ClickableText
+                        fontWeight={500}
+                        fontSize={14}
+                        // color={theme.text2}
```
Disabling the color prop.

biocom commented on src/custom/pages/Swap/SwapMod.tsx:
```diff
@@ -483,10 +487,20 @@ export default function Swap({
+                      <ClickableText
+                        fontWeight={500}
+                        fontSize={14}
+                        // color={theme.text2}
```
Disabling the color prop.

MareenG commented:
the link colour is different in the About section compared to the FAQ section 
<img width="639" alt="Bildschirmfoto 2021-05-14 um 12 45 20" src="https://user-images.githubusercontent.com/34510341/118260243-551bdd80-b4b2-11eb-9b21-de75bcc162e6.png">
<img width="648" alt="Bildschirmfoto 2021-05-14 um 12 45 07" src="https://user-images.githubusercontent.com/34510341/118260271-5ea54580-b4b2-11eb-9a76-7cba2473c98c.png">

biocom commented on src/pages/Swap/index.tsx:
```diff
@@ -383,7 +383,11 @@ export default function Swap({ history }: RouteComponentProps) {
+                      <Text
+                        fontWeight={500}
+                        fontSize={14}
+                        // color={theme.text2}
```
Disabling the color prop.

W3stside replied:
you need to rework this commented color prop as it's affecting way too many files and these aren't modded files either

biocom commented on src/pages/Swap/index.tsx:
```diff
@@ -395,10 +399,20 @@ export default function Swap({ history }: RouteComponentProps) {
+                      <ClickableText
+                        fontWeight={500}
+                        fontSize={14}
+                        // color={theme.text2}
```
Disabling the color prop.

biocom commented on src/pages/Swap/index.tsx:
```diff
@@ -395,10 +399,20 @@ export default function Swap({ history }: RouteComponentProps) {
+                      <ClickableText
+                        fontWeight={500}
+                        fontSize={14}
+                        // color={theme.text2}
```
Disabling the color prop.

biocom commented:
> In the About section the middle picture doesn't adjust to dark mode as the other two.
> <img alt="Bildschirmfoto 2021-05-14 um 12 42 45" width="647" src="https://user-images.githubusercontent.com/34510341/118259903-01a98f80-b4b2-11eb-8f06-281bb48641e5.png">

Yes, need to re-upload the image with a transparent background...

MareenG commented:
The name of the token list can't be seen 
<img width="551" alt="Bildschirmfoto 2021-05-14 um 12 48 02" src="https://user-images.githubusercontent.com/34510341/118260503-b2b02a00-b4b2-11eb-9a47-af27fd881cdc.png">

biocom reviewed (commented):
> the link colour is different in the About section compared to the FAQ section
> <img alt="Bildschirmfoto 2021-05-14 um 12 45 20" width="639" src="https://user-images.githubusercontent.com/34510341/118260243-551bdd80-b4b2-11eb-9b21-de75bcc162e6.png">
> <img alt="Bildschirmfoto 2021-05-14 um 12 45 07" width="648" src="https://user-images.githubusercontent.com/34510341/118260271-5ea54580-b4b2-11eb-9a76-7cba2473c98c.png">

Good find, updated!

biocom commented:
> The name of the token list can't be seen
> <img alt="Bildschirmfoto 2021-05-14 um 12 48 02" width="551" src="https://user-images.githubusercontent.com/34510341/118260503-b2b02a00-b4b2-11eb-9a47-af27fd881cdc.png">

Thanks for finding, fixed now! (waiting for a redeploy of the PR link).

MareenG commented:
The Pending Icon lost its colour
Before:
<img width="219" alt="Bildschirmfoto 2021-05-14 um 13 07 39" src="https://user-images.githubusercontent.com/34510341/118262629-703c1c80-b4b5-11eb-9151-0bc63014251a.png">
After: 
<img width="124" alt="Bildschirmfoto 2021-05-14 um 13 06 28" src="https://user-images.githubusercontent.com/34510341/118262640-7631fd80-b4b5-11eb-91dd-1a09a50276a4.png">
<img width="139" alt="Bildschirmfoto 2021-05-14 um 13 06 19" src="https://user-images.githubusercontent.com/34510341/118262643-77632a80-b4b5-11eb-9a69-f9194c8c0c7c.png">

MareenG commented:
I personally feel that the contrat of the box is not that good compared to the background. Maybe some form of border might improve this. 
<img width="619" alt="Bildschirmfoto 2021-05-14 um 13 10 25" src="https://user-images.githubusercontent.com/34510341/118262875-cc06a580-b4b5-11eb-994f-6fd65f255104.png">

> On some occasions there might be a black text where it should be a light text. (e.g. settings panel). 

I guess this applies also to this part :)

MareenG commented:
It is super cool I like it a lot :) 

I noticed that not every time the sun slides into the screen is this intended? It only does if I change the from and to token.

We probably don't want to get rid of the mooo but in the night mode an OWL hoo-h'HOO-hoo-hoo would be also quite fitting :owl:

biocom commented:
> It is super cool I like it a lot :)
> 
> I noticed that not every time the sun slides into the screen is this intended? It only does if I change the from and to token.
> 
> We probably don't want to get rid of the mooo but in the night mode an OWL hoo-h'HOO-hoo-hoo would be quite fitting for thr dark. mode 🦉

Yeah the SUN animation is only triggered once now. Will optimize these animations in a future PR.

Really like the idea about the night mode sound!!

biocom commented:
> I personally feel that the contrat of the box is not that good compared to the background. Maybe some form of border might improve this.
> <img alt="Bildschirmfoto 2021-05-14 um 13 10 25" width="619" src="https://user-images.githubusercontent.com/34510341/118262875-cc06a580-b4b5-11eb-994f-6fd65f255104.png">
> 
> > On some occasions there might be a black text where it should be a light text. (e.g. settings panel).
> 
> I guess this applies also to this part :)

Ah this upper part should be having a lighter BG. Thanks for checking! Will try to fix now.

MareenG commented:
I just noticed that it looks like the border is missing here for the input field (manage token list).

<img width="419" alt="Bildschirmfoto 2021-05-14 um 13 25 03" src="https://user-images.githubusercontent.com/34510341/118264401-ed689100-b4b7-11eb-81ae-941865414bac.png">

This is already in production

MareenG commented:
I just noticed that there is an Uniswap artefact in the Wallet address button and Change button when switching between wallets the border has the Uni pink. This is already in production.

<img width="90" alt="Bildschirmfoto 2021-05-14 um 13 33 17" src="https://user-images.githubusercontent.com/34510341/118265063-f7d75a80-b4b8-11eb-9f7e-ec7a957f5b35.png">
<img width="172" alt="Bildschirmfoto 2021-05-14 um 13 35 29" src="https://user-images.githubusercontent.com/34510341/118265326-4e449900-b4b9-11eb-95f5-64013ad974f0.png">

In the new design the button is pre highlighted before hovering over it

https://user-images.githubusercontent.com/34510341/118265259-29e8bc80-b4b9-11eb-88a4-ab0573991f95.mov

biocom commented:
> > I personally feel that the contrat of the box is not that good compared to the background. Maybe some form of border might improve this.
> > > On some occasions there might be a black text where it should be a light text. (e.g. settings panel).
> > 
> > 
> > I guess this applies also to this part :)
> 
> Ah this upper part should be having a lighter BG. Thanks for checking! Will try to fix now.

This should be fixed now.

biocom commented:
> The Pending Icon lost its colour
> Before:
> <img alt="Bildschirmfoto 2021-05-14 um 13 07 39" width="219" src="https://user-images.githubusercontent.com/34510341/118262629-703c1c80-b4b5-11eb-9151-0bc63014251a.png">
> After:
> <img alt="Bildschirmfoto 2021-05-14 um 13 06 28" width="124" src="https://user-images.githubusercontent.com/34510341/118262640-7631fd80-b4b5-11eb-91dd-1a09a50276a4.png">
> <img alt="Bildschirmfoto 2021-05-14 um 13 06 19" width="139" src="https://user-images.githubusercontent.com/34510341/118262643-77632a80-b4b5-11eb-9a69-f9194c8c0c7c.png">

This has now been addressed.

biocom commented:
> I just noticed that there is an Uniswap artefact in the Wallet address button and Change button when switching between wallets the border has the Uni pink. This is already in production.
> 
> <img alt="Bildschirmfoto 2021-05-14 um 13 33 17" width="90" src="https://user-images.githubusercontent.com/34510341/118265063-f7d75a80-b4b8-11eb-9f7e-ec7a957f5b35.png">
> 
> <img alt="Bildschirmfoto 2021-05-14 um 13 35 29" width="172" src="https://user-images.githubusercontent.com/34510341/118265326-4e449900-b4b9-11eb-95f5-64013ad974f0.png">
> 
> In the new design the button is pre highlighted before hovering over it
> 
>  Bildschirmaufnahme.2021-05-14.um.13.28.37.mov

Again good find. Updated the uni pink to use our orange.

alfetopito commented:
I love it!

Two tiny details:

![screenshot_2021-05-14_08-46-55](https://user-images.githubusercontent.com/43217/118295812-1cb7d780-b491-11eb-93f4-22ce061d9d49.png)
The price swap arrow on confirmation modal has a bad contrast

![screenshot_2021-05-14_08-45-55](https://user-images.githubusercontent.com/43217/118295857-2ccfb700-b491-11eb-8ea2-ed3ef60abbd5.png)
The default token image is blank.
On Rinkeby most tokens look bad.
Those will be the same in Mainnet if the token has no image.

W3stside commented on src/custom/components/Web3Status/index.tsx:
```diff
@@ -2,12 +2,24 @@ import React, { useMemo } from 'react'
+    background: ${({ theme }) => theme.wallet?.background};
+  }
+
+  ${Text} {
```
unused right? remove it

biocom replied:
Outdated comment (addressed).

W3stside commented on src/custom/components/swap/SwapModalFooter/SwapModalFooterMod.tsx:
```diff
@@ -56,8 +56,12 @@ export default function SwapModalFooter({
+            fontSize={14}
+            // color={theme.text2}
+          >
+            Pricesssssssssss
```
?

biocom replied:
Outdated comment (addressed).

W3stside commented on src/custom/pages/Swap/index.tsx:
```diff
@@ -47,34 +47,52 @@ const SwapModWrapper = styled(SwapMod)`
-      border-radius: 9px;
-      width: 28px;
-      height: 28px;
+      background: ${({ theme }) => theme.swap?.arrowDown?.background};
```
why are all these conditional?

biocom replied:
Made them conditional as it would leave out this property, if the `swap` and `arrowDown` are not provided in the theme. Which in case it falls back to the original (mod) properties (no override in that case was my thinking).

W3stside replied:
you're sure there's a fallback? it also seems a bit unnecessary to make such a specific nested colour. does `arrowDown` have many properties that are unique?

biocom replied:
Yes there's a fallback. Just updated now with removing the conditionals and these values are now mandatory (moved them to the baseTheme).

The reason for defining these values as theme properties, is that I wanted to make use of the `darkMode` value, without having to bring that over to the individual component. But instead make sure of the `darkMode` value in the theme value. Let me know your thoughts.

W3stside commented on src/custom/theme/cowSwapTheme.tsx:
```diff
@@ -25,45 +24,50 @@ export function colors(darkMode: boolean): Colors {
-    text3: darkMode ? '#000000' : '#000000',
-    text4: darkMode ? '#000000b8' : '#000000b8',
+    text1: darkMode ? '#c5daef' : '#000000',
+    text2: darkMode ? '#021E34' : '#000000',
```
why not re-enable all the `text2` code you commented out and make it work with a darkmode toggle here? why target specifically with CSS? all the commented color prop code was for `text2` no?

biocom replied:
The reason is that, in comparison with the UNI front-end, on our end it's not a 1-on-1 translation in terms of switching out colors: On several occasions we use a different color contrast and thus our existing variables don't match that scenario (e.g. resulting in a dark text on a dark background).

W3stside commented on src/custom/theme/cowSwapTheme.tsx:
```diff
@@ -162,7 +195,7 @@ export function theme(darkmode: boolean): DefaultTheme {
     // Overide Theme
     ...baseThemeVariables(darkmode, colorsTheme),
-    ...themeVariables(colorsTheme)
+    ...themeVariables(darkmode, colorsTheme)
```
switch the param order around if you're going to do this to keep it as similar as Uni's

biocom replied:
Not sure I understand: Didn't we already have this order in place, looking at `...baseThemeVariables(darkmode, colorsTheme),` ? Didn't see direct examples on the original UNI code.

W3stside replied:
theme Variables is our code? Then all good,, thought it was a function coming from Uni we extended

biocom commented:
@W3stside 
> you need to rework this commented color prop as it's affecting way too many files and these aren't modded files either

Thanks for your review. I'll give it another go at trying to override with CSS. In case that fails on my end, my thinking is, any modification/addition to the MOD file to make it work, likely causes adding more code than simply commenting it out?

Is commenting out something in the MOD file not possible in our scenario at all? My initial understand was that it was the mere introduction of *new* or *modified* code that we'd want to prevent.

W3stside commented:
> @W3stside
> 
> > you need to rework this commented color prop as it's affecting way too many files and these aren't modded files either
> 
> Thanks for your review. I'll give it another go at trying to override with CSS. In case that fails on my end, my thinking is, any modification/addition to the MOD file to make it work, likely causes adding more code than simply commenting it out?
> 
> Is commenting out something in the MOD file not possible in our scenario at all? My initial understand was that it was the mere introduction of _new_ or _modified_ code that we'd want to prevent.

I just feel that it can be fixed at the source rather than trying to remove the prop from all the ones using Text. or just mod that component and import it where we need it in the mod files. but if @anxolin and @alfetopito don't care then i can hop on board and say yeah take the frictionless approach of changing without modding since it's minor.

biocom commented:
@W3stside I tried addressing all your comments. Primarily reverted direct mods. These files still have mods:
- `src/custom/components/Settings/SettingsMod.tsx`
  - addition of `color={theme.text2}`
- `src/custom/components/TransactionConfirmationModal/TransactionConfirmationModalMod.tsx `
   - comment out `color` prop)
- `src/custom/components/swap/SwapModalFooter/SwapModalFooterMod.tsx `
  - addition of `color={theme.text2}`
- `src/custom/components/swap/SwapModalHeader/SwapModalHeaderMod.tsx `
  - comment out `color={theme.text2}`
  - addition of `color={theme.text1}`)
  
What they all have in common is the `Modal` component. I tried overriding these values from the parent components, that calls the Modal. But the Modal is not a direct child of the parent component. What would you advise?
  - Leave mods as is
  - Introduce a `className` prop to the modal so we can identify specific Modals. In that case we'd have to add styles for it in a globalStyle type of environment.
  - Something else.

biocom commented:
@W3stside Now refactored to not touch any mod files. Please review.

W3stside commented on src/custom/components/TransactionSettings/index.tsx:
```diff
@@ -17,6 +18,15 @@ const Option = styled(FancyButtonUni)<{ active: boolean }>`
 `
 
+const Wrapper = styled.div`
+  ${OptionCustom} {
```
is it not possible to override OptionCustom with styled? by adding className prop should work no?

biocom replied:
Probably yes, but to me this looks the least obtrusive without adding stuff to the Mod file. Think I should modify?

W3stside commented on src/custom/theme/cowSwapTheme.tsx:
```diff
@@ -16,6 +14,13 @@ import {
+import { cowSwapBackground, cowSwapLogo } from './cowSwapAssets'
+
+// Modal override items
+import { HeaderText } from '@src/components/WalletModal/Option'
```
this path doesn't need @src or does it?

biocom replied:
My understanding is, that `@src` pulls the original component, where we export `HeaderText`. That's the one I want to target and override. We don't have a `custom` component with a HeaderText.

W3stside replied:
yup then fine

anxolin commented:
Great progress since my last visit to this PR. It looks great.

Tiny details I saw. Imo the wrapper of ETH has a strange background. It should be more coherent with the light mode:

Right now looks like this:
![image](https://user-images.githubusercontent.com/2352112/118711636-1c686500-b820-11eb-9023-fc69920bf25d.png)

I suggest colors like this:
<img width="677" alt="Screenshot at May 18 21-27-19" src="https://user-images.githubusercontent.com/2352112/118711702-2c804480-b820-11eb-9b2c-d1323dae1564.png">

Also, the button, Wrap my Ether, has a color that don't fit in.  Not sure what is the best color, but for example imo this other color looks better for it:
![image](https://user-images.githubusercontent.com/2352112/118711980-83861980-b820-11eb-90ce-d3063083bc22.png)

anxolin commented:
I'm very bad at color picking, but I'll give my probably wrong view.

I feel this light blue works for texts in dark mode, but is not good for backgrounds. 
![image](https://user-images.githubusercontent.com/2352112/118713511-4f135d00-b822-11eb-9e5c-649ec953b5ec.png)

Imo, we need a EVEN DARKER blue/grayish for some places where we use this.

For example:
![image](https://user-images.githubusercontent.com/2352112/118713626-736f3980-b822-11eb-8c24-1771cc29ab5b.png)


![image](https://user-images.githubusercontent.com/2352112/118714421-deb90b80-b822-11eb-9f34-4288778cbd79.png)

biocom commented:
@anxolin Thanks for the feedback. Will try to address in this PR. First part the settings panel:
<img width="350" alt="Screen Shot 2021-05-19 at 10 50 18" src="https://user-images.githubusercontent.com/31534717/118784207-1ca04900-b890-11eb-9e04-aaa60acbac27.png"><img width="350" alt="Screen Shot 2021-05-19 at 10 46 16" src="https://user-images.githubusercontent.com/31534717/118784224-1f02a300-b890-11eb-913a-c21b206f742b.png">

biocom commented:
<img width="350" alt="Screen Shot 2021-05-19 at 11 03 43" src="https://user-images.githubusercontent.com/31534717/118786322-fe3b4d00-b891-11eb-96ef-458ce78e6cbd.png"><img width="350" alt="Screen Shot 2021-05-19 at 11 03 34" src="https://user-images.githubusercontent.com/31534717/118786327-ff6c7a00-b891-11eb-8f85-a2f2990d4435.png">

anxolin commented:
> First part the settings panel
Looks great!

W3stside reviewed (approved):
good!

alfetopito reviewed (approved):
Awesome work :+1: 

One thing that I noticed and likely not from this PR, thus to be addressed separately, is that there are two "swap" arrows when expert mode is on and add recipient is clicked
![screenshot_2021-05-19_10-54-38](https://user-images.githubusercontent.com/43217/118860973-d7304b80-b890-11eb-92eb-01eb0f98bf69.png)

Clicking on the second arrow has no effect
